src/main/java/reposense/parser/CsvParser.java

```diff
@@ -51,105 +52,113 @@ public CsvParser(Path csvFilePath) throws IOException {
-     * value in {@code position} is not empty.
+     * Returns the value of {@code record} at {@code colNum}.
      */
-    private boolean containsValueAtPosition(final String[] elements, int position) {
-        return elements.length > position && !elements[position].isEmpty();
+    protected String get(final CSVRecord record, int colNum) {
+        return record.get(colNum).trim();
     }
 
     /**
-     * Removes the override keyword for {@code position} in {@code elements}.
+     * Returns the value of {@code record} at {@code colNum} if present, or
+     * returns {@code defaultValue} otherwise.
      */
-    protected void removeOverrideKeywordFromElement(final String[] elements, int position) {
-        if (isElementOverridingStandaloneConfig(elements, position)) {
-            elements[position] = elements[position].replaceFirst(OVERRIDE_KEYWORD, "");
-        }
+    protected String getOrDefault(final CSVRecord record, int colNum, String defaultValue) {
+        return get(record, colNum).isEmpty() ? defaultValue : get(record, colNum);
     }
 
     /**
-     * Gets the value of {@code position} in {@code elements}.
-     * Returns the value of {@code position} if it is in {@code element} and not empty.
-     * Otherwise returns an empty string.
+     * Returns the value of {@code record} at {@code colNum} as a {@code List},
+     * delimited by {@code COLUMN_VALUES_SEPARATOR} if it is in {@code record} and not empty, or
+     * returns an empty {@code List} otherwise.
      */
-    protected String getValueInElement(final String[] elements, int position) {
-        return (containsValueAtPosition(elements, position)) ? elements[position] : "";
+    protected List<String> getAsList(final CSVRecord record, int colNum) {
+        if (get(record, colNum).isEmpty()) {
+            return Collections.emptyList();
+        }
+        return Arrays.stream(get(record, colNum).split(COLUMN_VALUES_SEPARATOR))
+                .map(String::trim)
+                .collect(Collectors.toList());
     }
 
     /**
-     * Gets the value of {@code position} in {@code elements}.
-     * Returns the value of {@code position} if it is in {@code element} and not empty.
-     * Otherwise returns the {@code defaultValue}.
+     * Returns the values in {@code record} as a list with the {@link CsvParser#OVERRIDE_KEYWORD} prefix removed.
+     * Returns an empty list if {@code record} at {@code colNum} is empty.
      */
-    protected String getValueInElement(final String[] elements, int position, String defaultValue) {
-        return (containsValueAtPosition(elements, position)) ? elements[position] : defaultValue;
+    protected List<String> getAsListWithoutOverridePrefix(final CSVRecord record, int colNum) {
+        List<String> data = getAsList(record, colNum);
+        if (isElementOverridingStandaloneConfig(record, colNum)) {
+            data.set(0, data.get(0).replaceFirst(OVERRIDE_KEYWORD, ""));
+        }
+        data.removeIf(String::isEmpty);
```

Comment:
The only way to have an empty string is caused by the replacement in line 135?

If so, its better to move this line to the if block,

Reply:
Made the changes